config/v1/types_dns.go

```diff
@@ -20,6 +20,9 @@ type DNS struct {
 }
 
 type DNSSpec struct {
+	// BaseDomain is the base domain of the cluster. All managed DNS records will
+	// be sub-domains of this base.
+	BaseDomain string `json:"baseDomain"`
```

Comment:
This might be useful in Infra ?

Reply:
Yeah, this is probably a higher level config setting.  Both DNS and the IngressOperator need it.  I am not sure if anyone else wants to know, so I don't know if keeping it in DNS is sufficient for now.

Reply:
Is there supposed to be any correspondence between config object and operator? Logically DNS-related configuration belongs here: "DNS holds cluster-wide information about DNS." That doesn't specifically mean config for the DNS operator.

Reply:
@Miciah shares my interpretation of this type. If the base domain (the only required DNS input to the installer?) I'm not sure what does.

Reply:
> Yeah, this is probably a higher level config setting. Both DNS and the IngressOperator need it. I am not sure if anyone else wants to know, so I don't know if keeping it in DNS is sufficient for now.

As a cluster-admin, if I want to change my DNS suffix, where would I expect to do it?  The types here are not mapped to operators, but instead to features. The feature here being DNS.

The ingress operator is perfectably able to consume the information from the DNS config type if it needs it as an input.